### PR TITLE
Fix Viewer Crashing on Load of Wrong OSCAL Doc Type

### PR DESCRIPTION
**Pull request by zclarkEDC**
Fix for [#240 ](https://github.com/EasyDynamics/oscal-react-library/issues/240)

Should work for Catalogs, SSPs, and Components.

### CODE REVIEW

**Comment by kylelaker** [comment]
Should we move `ErrorBoundary` into it's own file and export it so that we can have other error boundaries all over?

**Comment by kylelaker** [comment]
That would probably also mean that `ErrorBoundary` should accept an `errorFormatter` prop or something like that so that you can override how an error is displayed at a specific part of the application:

snippets of changed functions:

```js
function defaultFormatter(error) {
  return (
    <Alert severity="error">
      Yikes! Something went wrong loading the OSCAL data. Sorry, we&apos;ll look
      into it. ({error.message})
    </Alert>
  );
);

export class ErrorBoundary extends React.Component {
  constructor(props) {
    super(props)
    this.errorFormatter = props.errorFormatter ?? defaultFormatter;
    this.state = { error: undefined };
  }

  render() {
    if (this.state.error !== undefined) {
      return this.errorFormatter(this.state.error);
    }
    return this.props.children;
  }
}
```

Idk if the `errorFormatter` should go on `this` or `this.state`. I also don't know if I like that name.

File: src/components/OSCALLoader.js [hunk 1]
```diff
@@ -99,8 +99,30 @@ function restPatch(
...
+    this.state = { error: undefined };
+  }
+
+  static getDerivedStateFromError(error) {
+    return { error };
+  }
+
+  componentDidCatch(error, errorInfo) {
+    this.setState({ error });
+    console.error({ ...this.state, errorInfo });
```
Comment (kylelaker) [inline src/components/OSCALLoader.js:114]: We probably also want to remove this

Comment (zclarkEDC) [reply src/components/OSCALLoader.js:114]: Yeah I was wondering why that was removed

Comment (zclarkEDC) [reply src/components/OSCALLoader.js:114]: Actually it doesn't work without `componentDidCatch`,

File: src/components/ErrorBoundary.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+export function defaultFormatter(error) {
+  return (
+    <Alert severity="error">
+      Yikes! Something went wrong loading the OSCAL data. Sorry, we&apos;ll look
+      into it. ({error.message})
+    </Alert>
+  );
+}
+
+export class ErrorBoundary extends React.Component {
```
Comment (kylelaker) [inline src/components/ErrorBoundary.js:13]: Should we actually `export default` to match the style of the other components?

File: src/components/ErrorBoundary.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
+import React from "react";
+import Alert from "@material-ui/lab/Alert";
+
+export function defaultFormatter(error) {
```
Comment (kylelaker) [inline src/components/ErrorBoundary.js:4]:
```suggestion
function defaultFormatter(error) {
```

Let's _not_ export this

**Review by kylelaker — CHANGES_REQUESTED** [review]

**Review by kylelaker — APPROVED** [review]

File: src/components/ErrorBoundary.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+  componentDidCatch(error) {
+    this.setState({ error });
+  }
+
+  render() {
+    if (this.state.error !== undefined) {
+      return this.errorFormatter(this.state.error);
+    }
+    return this.props.children;
```
Comment (pkothare) [inline src/components/ErrorBoundary.js:28]:
```suggestion
    return this.props.children || null;
```
The reason why is that `this.props` or `this.props.children` can be undefined, and I was receiving the error:

```
Uncaught Error: ErrorBoundary(...): Nothing was returned from render. This usually means a return statement is missing. Or, to render nothing, return null.
```

**Review by pkothare — CHANGES_REQUESTED** [review]

File: src/components/ErrorBoundary.js (hunk 4 above)
Comment (zclarkEDC) [reply src/components/ErrorBoundary.js:28]: Good catch!

**Review by pkothare — APPROVED** [review]

**Pull request merged**